modules/api-compatibility-guidelines.adoc

```diff
@@ -13,6 +13,7 @@ Red Hat recommends that application developers adopt the following principles in
 * Do not design applications that rely on configuration files provided by system packages or other components. These files can change between versions unless the upstream community is explicitly committed to preserving them. Where appropriate, depend on any Red Hat provided interface abstraction over those configuration files in order to maintain forward compatibility. Direct file system modification of configuration files is discouraged, and users are strongly encouraged to integrate with an Operator provided API where available to avoid dual-writer conflicts.
 * Do not depend on API fields prefixed with `unsupported<FieldName>` or annotations that are not explicitly mentioned in product documentation.
 * Do not depend on components with shorter compatibility guarantees than your application.
+* Do not interact directly with the etcd server, all etcd access must be performed via the api-server.
```

Comment:
Even for backups and restores operations? 

IE: https://docs.openshift.com/container-platform/4.9/backup_and_restore/index.html

I assume this is excluded; but 'direct access' or 'interaction' is required for these procedures.

Reply:
good catch, no the thing we're trying to exclude here is people making direct create/update calls to store their own objects in etcd (or, worse, updating ours)

Reply:
@sferich888 updated, see if that looks better

Reply:
(note that i've shoehorned this into the "application developer" guidance.  i didn't see a better place to document this and i didn't want to create a whole new one for this item)